### PR TITLE
generate cmd: fix reading from stdin, add test

### PR DESCRIPTION
**Pull request by andreasgerstmayr**

### CODE REVIEW

File: cmd/generate/main_test.go [hunk 1]
```diff
@@ -63,3 +65,46 @@ metadata:
...
+
+	c.SetArgs([]string{"generate"})
+	_, err := c.ExecuteC()
+	require.NoError(t, err)
+
+	require.Contains(t, out.String(), `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  creationTimestamp: null
```
Comment (iblancasa) [inline cmd/generate/main_test.go:101]: Maybe: should we fix this in the future?  A `null` timestamp sounds weird for me

Comment (andreasgerstmayr) [reply cmd/generate/main_test.go]: Done

**Review by iblancasa — COMMENTED** [review]
LGTM. It seems the CI needs some fixes

**Comment by andreasgerstmayr** [comment]
CI passes, codecov just complains that the file output path isn't tested. unit tests writing to the filesystem can be problematic, therefore imho it's still good to merge.

**Comment by andreasgerstmayr** [comment]
codecov is now happy :)

**Review by iblancasa — COMMENTED** [review]
Good job!
It seems the coverage is not happy again

**Review by frzifus — APPROVED** [review]
Iam fine with ignoring codecov here :baby_chick:

**Pull request merged**